e2e/test/scenarios/sharing/reproductions/26988-embedding-dynamic-settings.cy.spec.js

```diff
@@ -0,0 +1,56 @@
+import {
+  describeEE,
+  getIframeBody,
+  popover,
+  restore,
+  setTokenFeatures,
+  visitDashboard,
+} from "e2e/support/helpers";
+import { ORDERS_ID } from "metabase-types/api/mocks/presets";
+
+describeEE("issue 26988", () => {
+  beforeEach(() => {
+    restore();
+    cy.intercept("GET", "/api/embed/dashboard/*").as("dashboard");
+
+    cy.signInAsAdmin();
+    setTokenFeatures("all");
+  });
+
+  it("should apply embedding settings passed in URL on load", () => {
+    cy.createQuestionAndDashboard({
+      questionDetails: {
+        name: "Q1",
+        query: {
+          "source-table": ORDERS_ID,
+          limit: 3,
+        },
+      },
+      dashboardDetails: {
+        enable_embedding: true,
+      },
+    }).then(({ body: card }) => {
+      visitDashboard(card.dashboard_id);
+    });
+
+    cy.icon("share").click();
+    cy.findByRole("button", { name: "Set up" }).click();
+    cy.wait("@dashboard");
+
+    getIframeBody().should("have.css", "font-family", `Lato, sans-serif`);
+
+    cy.findByTestId("embedding-settings").findByText("Font").siblings().click();
```

Comment:
Let's make this a bit more readable

```suggestion
    cy.findByTestId("embedding-settings").findByLabelText("Font").click();
```

But you have to modify the markup a bit as well.

Here's what I did to the [`DisplayOptionSection`](https://github.com/metabase/metabase/blob/6fbc94fe67463ec292b995416c5eb602659dcb93/frontend/src/metabase/public/components/widgets/DisplayOptionsPane.jsx#L129-L134), but maybe you could come up with something better.

```ts
const DisplayOptionSection = ({ title, children }) => {
  const uniqueId = useUniqueId("display-option");
  return (
    <DisplayOption>
      <DisplayOptionTitle id={uniqueId}>{title}</DisplayOptionTitle>
      <span aria-labelledby={uniqueId}>{children}</span>
    </DisplayOption>
  );
};
```